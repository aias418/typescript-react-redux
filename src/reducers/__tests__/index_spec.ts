import reducers, {
  initialState,
} from '../index'

import {
  Action,
  incrementCounter
} from '../../actions'

const resultOf = (actions: Action[]) =>
  actions.reduce(reducers, initialState)

describe.only('reducers/counter', () => {
  it('starts at 0', () =>
    expect(resultOf([])).toMatchSnapshot())

  it('increments', () =>
    expect(resultOf([
      incrementCounter(1),
      incrementCounter(2),
      incrementCounter(3),
    ])).toMatchSnapshot())

  it('restores state', () =>
    expect(resultOf([
      {
        type: 'LOAD_COUNT_SUCCESS',
        request: {},
        response: { value: 14 },
      }
    ])).toMatchSnapshot())
})
