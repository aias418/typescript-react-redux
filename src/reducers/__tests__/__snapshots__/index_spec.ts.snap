// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/counter increments 1`] = `
Object {
  "counter": Object {
    "value": 6,
  },
  "error": "",
  "isLoading": false,
  "isSaving": false,
}
`;

exports[`reducers/counter restores state 1`] = `
Object {
  "counter": Object {
    "value": 14,
  },
  "error": "",
  "isLoading": false,
  "isSaving": false,
}
`;

exports[`reducers/counter starts at 0 1`] = `
Object {
  "counter": Object {
    "value": 0,
  },
  "error": "",
  "isLoading": false,
  "isSaving": false,
}
`;
